actionpack/lib/action_controller/railtie.rb

```diff
@@ -24,7 +24,7 @@ class Railtie < Rails::Railtie #:nodoc:
     initializer "action_controller.parameters_config" do |app|
       options = app.config.action_controller
 
-      ActiveSupport.on_load(:action_controller) do
+      ActiveSupport.on_load(:action_controller_base) do
```

Comment:
So the problem is because we run this block twice and after the first one we delete the values, so the second one we get the default values (which is false).

Yeah, we still need to make sure it will run for `ActionController::API`.

I think we need an new concept that are lazy load hooks that are only executed once. Something like:

```
ActiveSupport.on_load(:action_controller_base, only_once: true) do
```

Could you implement that?